### PR TITLE
GroupExScheduleFetcher cache not validated correctly #1928

### LINKED ISSUE #1928: GroupExScheduleFetcher cache not validated correctly
When a user visits a page using the Groupex Pro `group_schedules` block, once they've made the initial selection, changing the form does not update the results.

After some debugging, I discovered that by commenting out
```
 if ($this->schedule) {
   return $this->schedule;
 }  
```
line 127 in `docroot/profiles/contrib/openy/modules/custom/openy_group_schedules/src/GroupexScheduleFetcher.php` the results will update correctly.

Suggestion: we need to check that the parameters are the same for the new query, and not just that the schedule exists.

### PR DESCRIPTION
**Pull request by rahulusagility**
Original Issue, this PR is going to fix: https://github.com/ymcatwincities/openy/issues/1928

Make sure these boxes are checked before asking for review of your pull request - thank you!

If there is a new feature or this is a bug fix - use 8.x-2.x branch. We'll tag for release if the bug is critical asap or tag for release next bug fix release until critical issue arrived.

## Steps for review

- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer. See [Code of Conduct](https://github.com/ymcatwincities/openy/wiki/Open-Y-Code-of-Conduct-and-Best-Practices).
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

File: modules/custom/openy_group_schedules/src/GroupexScheduleFetcher.php [hunk 1]
```diff
@@ -124,9 +124,9 @@ public function __construct(GroupexHelper $groupex_helper, ConfigFactoryInterfac
    */
   public function getSchedule() {
     // Use cached schedule if already processed.
-    if ($this->schedule) {
-      return $this->schedule;
-    }
+    // if ($this->schedule) {
+    //   return $this->schedule;
+    // }
```
Comment (podarok) [inline modules/custom/openy_group_schedules/src/GroupexScheduleFetcher.php:129]: we can't keep commented code, as per Drupal code quality requirements. 
If we need to have code disable - let's introduce settings/option in Admin UI
If code would be never used - let's just remove it

**Review by podarok — CHANGES_REQUESTED** [review]
Let's remove commented code, please

**Comment by mattshoaf** [comment]
Hi everyone, I presented commented out code in the related issue as a workaround for the Y site that we work with to give a clue as to where this was breaking. I think that if we dig into the `$this->schedule` array in the if conditional, we can find the appropriate parameter to check, something like `if($this->schedule['params'] == $submitted_form_params)`

My opinion: I would rather find a fix for this, if we remove caching completely, there would be a lot of other code to depreciate besides just these lines. Also, it would be ideal to fix this so that we don't have to ping the GroupEx servers for identical queries.

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Comment by podarok** [comment]
I'm going to close this because of low demand
This PR is just an example of a workaround.

**Pull request closed without merging**